src/jit/gentree.cpp

```diff
@@ -7569,14 +7611,14 @@ GenTreePtr Compiler::gtCloneExpr(
             case GT_IND:
             case GT_OBJ:
             case GT_STORE_OBJ:
-                if (tree->gtFlags & GTF_IND_ARR_INDEX)
+            {
+                ArrayInfo arrInfo;
+                if (TryGetArrayInfo(tree->AsIndir(), &arrInfo) && !tree->AsIndir()->gtOp1->OperIs(GT_INDEX_ADDR))
```

Comment:
Did you consider passing a argument stating whether we should cons up an ArrayInfo for the GT_INDEX_ADDR case.
Here we cons it up and discard it.